AsyncDisplayKit/ASCollectionView.h

```diff
@@ -342,18 +353,34 @@
  */
 - (BOOL)shouldBatchFetchForCollectionView:(ASCollectionView *)collectionView;
 
+@end
+
+@protocol ASCollectionViewDelegateFlowLayout <ASCollectionViewDelegate>
+
+@optional
+
 /**
  * Passthrough support to UICollectionViewDelegateFlowLayout sectionInset behavior.
  *
  * @param collectionView The sender.
  * @param collectionViewLayout The layout object requesting the information.
- * #param section The index number of the section whose insets are needed.
+ * @param section The index number of the section whose insets are needed.
  *
  * @discussion The same rules apply as the UICollectionView implementation, but this can also be used without a UICollectionViewFlowLayout.
  * https://developer.apple.com/library/ios/documentation/UIKit/Reference/UICollectionViewDelegateFlowLayout_protocol/index.html#//apple_ref/occ/intfm/UICollectionViewDelegateFlowLayout/collectionView:layout:insetForSectionAtIndex:
  *
  */
-- (UIEdgeInsets)collectionView:(UICollectionView *)collectionView layout:(UICollectionViewLayout *)collectionViewLayout insetForSectionAtIndex:(NSInteger)section;
+- (UIEdgeInsets)collectionView:(ASCollectionView *)collectionView layout:(UICollectionViewLayout *)collectionViewLayout insetForSectionAtIndex:(NSInteger)section;
+
+/**
+ * Asks the delegate for the size of the header in the specified section.
+ */
+- (CGSize)collectionView:(ASCollectionView *)collectionView layout:(UICollectionViewLayout *)collectionViewLayout referenceSizeForHeaderInSection:(NSInteger)section;
```

Comment:
Let me know as your thinking evolves on this.  In particular, I'd like to understand from your use case what the practical adoption considerations are for an app that is in fact using ASDK (or will be using ASDK) — does it provide any advantage to have these methods besides the familiarity?  Perhaps it does, if there is some kind of custom size logic that would otherwise have to be moved into a node subclass, but I am completely without experience using supplementary views and making nontrivial UICollectionViewLayout subclasses.